src/candid_rpc.rs

```diff
@@ -27,34 +30,43 @@ impl RpcTransport for CanisterTransport {
         METADATA.with(|m| m.borrow().get().nodes_in_subnet)
     }
 
-    fn resolve_api(provider: RpcNodeProvider) -> Result<RpcApi, ProviderError> {
-        // TODO: https://github.com/internet-computer-protocol/ic-eth-rpc/issues/73
-        Ok(provider.api())
+    fn resolve_api(provider: &RpcNodeProvider) -> Result<RpcApi, ProviderError> {
+        use RpcNodeProvider::*;
+        let (chain_id, hostname) = match provider {
+            Ethereum(provider) => (
+                ETH_MAINNET_CHAIN_ID,
+                match provider {
+                    EthereumProvider::Ankr => "rpc.ankr.com",
+                    EthereumProvider::BlockPi => "ethereum.blockpi.network",
```

Comment:
FYI @rvanasa : We are going to kick out BlockPi on the used providers for ckETH because it (without API keys) it has a hard limit of 25kB on the response size of any request that was problematic for us (`eth_getLogs`)